Kinetic_shape_reconstruction/include/CGAL/Kinetic_shape_partition_3.h

```diff
@@ -0,0 +1,3425 @@
+
+  \param face_index
+   index of the query face.
+
+  \param it
+  output iterator.
+
+  \pre successful partition
+  */
+  template<class OutputIterator>
+  void exact_vertices(const Index& face_index, OutputIterator it) const {
+
+    for (auto& p : m_partition_nodes[face_index.first].face2vertices[face_index.second])
+      *it++ = m_partition_nodes[p.first].m_data->exact_vertices()[p.second];
+  }
+
+  /*!
+  \brief Vertices and their exact positions of a face of the kinetic partition.
+
+  \tparam OutputIterator
+  must be an output iterator to which `IntersectionTraits::Point_3` can be assigned.
+
+  \tparam IndexOutputIterator
+  must be an output iterator to which `Index` can be assigned.
+
+  \param face_index
+   index of the query face.
+
+  \param pit
+  output iterator for vertex positions.
+
+  \param iit
+  output iterator for vertices.
+
+  \pre successful partition
+  */
+  template<class OutputIterator, class IndexOutputIterator>
+  void exact_vertices(const Index& face_index, OutputIterator pit, IndexOutputIterator iit) const {
+    const auto& v = m_partition_nodes[face_index.first].m_data->exact_vertices();
+    for (auto& i : m_partition_nodes[face_index.first].m_data->face_to_vertices()[face_index.second]) {
+      *iit++ = std::make_pair(face_index.first, i);
+      *pit++ = v[i];
+    }
+  }
+
+  /*!
+  \brief Face indices of a volume.
+
+  \tparam OutputIterator
+  must be an output iterator to which `Index` can be assigned.
+
+  \param volume_index
+   index of the query volume.
+
+  \param it
+  output iterator.
+
+  \pre successful partition
+  */
+  template<class OutputIterator>
+  void faces(std::size_t volume_index, OutputIterator it) const {
+    CGAL_assertion(m_volumes.size() > volume_index);
+    auto p = m_volumes[volume_index];
+
+    for (std::size_t i : m_partition_nodes[p.first].m_data->volumes()[p.second].faces)
+      *it++ = std::make_pair(p.first, i);
+  }
+
+  /*!
+  \brief Retrieves all unique faces of the partition. Unique means that the shared face between two adjacent volumes is only contained once.
+
+  \tparam OutputIterator
+  must be an output iterator to which `Index` can be assigned.
+
+  \param it
+  output iterator.
+
+  \pre successful partition
+  */
+  template<class OutputIterator>
+  void unique_faces(OutputIterator it) const {
+    for (std::size_t i = 0; i < m_partition_nodes.size(); i++) {
+      const Sub_partition& p = m_partition_nodes[i];
+      for (std::size_t j = 0; j < p.face_neighbors.size(); j++) {
+        if (p.face_neighbors[j].second.first == i)
+          *it++ = Index(i, j);
+        else if (i < p.face_neighbors[j].second.first)
+          *it++ = Index(i, j);
+      }
+    }
+  }
+
+  /*!
+  \brief Indices of adjacent volumes. Negative indices correspond to the empty spaces behind the sides of the bounding box.
+
+    -1 zmin\n
+    -2 ymin\n
+    -3 xmax\n
+    -4 ymax\n
+    -5 xmin\n
+    -6 zmax\n
```

Comment:
Probably a better alternative:
```
 1. zmin
 2. ymin
 3. xmax
 4. ymax
 5. xmin
 6. zmax
```
resulting in:
![image](https://github.com/CGAL/cgal/assets/5223533/32ea1333-562a-430b-908a-b9b50642e550)

Reply:
Thank you, but I am specifically linking negative numbers with axis directions.

Reply:
Ah understood, maybe placing backticks around the numbers like `` `-1` zmin\n ``